src/boot/bootctl-uki.c

```diff
@@ -30,12 +30,19 @@ static int pe_sections(FILE *uki, struct PeSectionHeader **ret, size_t *ret_n) {
         assert(ret);
         assert(ret_n);
 
-        items = fread(&dos, 1, sizeof(dos), uki);
-        if (items != sizeof(dos))
-                return log_error_errno(SYNTHETIC_ERRNO(EIO), "DOS header read error");
+        items = fread(&dos, 1, sizeof(dos.Magic), uki);
```

Comment:
to make this less surprising, maybe change first argument to `&dos.Magic`, too.

might confuse static checkers otherwise.

Reply:
or alternatively:

```c
items = fread(&dos, 1, endoffsetof_field(DosFileHeader, Magic), uki);
```

Reply:
and of course the change the items == check below also to use endoffsetof_field()